kolibri/content/api.py

```diff
@@ -305,23 +305,8 @@ def filter_by_role(self, queryset, name, value):
             if user.roles.exists() or user.is_superuser:  # must have coach role or higher
                 return queryset
 
-        # In all other cases, exclude leaf nodes that are coach content
-        queryset = queryset.exclude(coach_content=True)
-
-        # Also exclude topics that are 100% coach content
-        def node_only_has_coach_content(contentnode):
-            if (contentnode.kind == content_kinds.TOPIC):
-                return not contentnode.get_descendants() \
-                    .exclude(kind=content_kinds.TOPIC) \
-                    .exclude(coach_content=True) \
-                    .exclude(available=False) \
-                    .exists()
-            else:
-                return contentnode.coach_content
-
-        only_coach_content_node_ids = [node.id for node in queryset if node_only_has_coach_content(node)]
-
-        return queryset.exclude(id__in=only_coach_content_node_ids)
+        # In all other cases, exclude nodes that are coach content
+        return queryset.exclude(coach_content=True)
```

Comment:
Are there implications also for when we show those little blue CC icons?

Seems as if there are two different things we want to know: is there *any* CC content under this node, and is *all* the content under this node CC?

Reply:
This is for filtering by role, I didn't touch the `num_coach_contents` field on the serializer which is used for the coach contents icons.